Library/Homebrew/extend/os/linux/extend/ENV/shared.rb

```diff
@@ -7,8 +7,10 @@ def effective_arch
       @bottle_arch.to_sym
```

Comment:
If we go with the approach I mentioned about dropping `native` entirely: can this whole file get removed?

Reply:
Yes but it would still require the fixes elsewhere (e.g. `arch_flag`) to make it work with `Hardware.oldest_cpu` so probably won't actually save any code for this particular PR.